File: 2048_8bit/core/src/com/nopalsoft/dosmil/screens/MainMenuScreen.java
package com.nopalsoft.dosmil.screens;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input.Keys;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.badlogic.gdx.utils.Align;
import com.nopalsoft.dosmil.Assets;
import com.nopalsoft.dosmil.MainGame;
import com.nopalsoft.dosmil.Settings;
import com.nopalsoft.dosmil.game.GameScreen;

public class MainMenuScreen extends Screens {

    Image titleImage;

    Label labelPlay;
    Label labelHelp;
    Label labelLeaderboard;
    Label labelRate;

    Button buttonMusic;
    Button buttonSound;
    Button buttonFacebook;

    public MainMenuScreen(final MainGame game) {
        super(game);
        titleImage = new Image(Assets.titleAtlasRegion);
        titleImage.setPosition(SCREEN_WIDTH / 2f - titleImage.getWidth() / 2f, 580);

        labelPlay = new Label(Assets.languagesBundle.get("play"), Assets.labelStyleLarge);
        labelPlay.setPosition(SCREEN_WIDTH / 2f - labelPlay.getWidth() / 2f, 450);
        addPressEffect(labelPlay);
        labelPlay.addListener(new ClickListener() {
            public void clicked(InputEvent event, float x, float y) {
                changeScreenWithFadeOut(GameScreen.class, game);
            }
        });

        // Help
        labelHelp = new Label(Assets.languagesBundle.get("help"), Assets.labelStyleLarge);
        labelHelp.setPosition(SCREEN_WIDTH / 2f - labelHelp.getWidth() / 2f, 350);
        addPressEffect(labelHelp);
        labelHelp.addListener(new ClickListener() {
            public void clicked(InputEvent event, float x, float y) {
                changeScreenWithFadeOut(HelpScreen.class, game);
            }
        });

        // Rate
        labelRate = new Label(Assets.languagesBundle.get("rate"), Assets.labelStyleLarge);
        labelRate.setPosition(SCREEN_WIDTH / 2f - labelRate.getWidth() / 2f, 250);
        addPressEffect(labelRate);

        // Leaderboard
        labelLeaderboard = new Label(Assets.languagesBundle.get("leaderboard"), Assets.labelStyleLarge);
        labelLeaderboard.setFontScale(.85f);
        labelLeaderboard.setWidth(SCREEN_WIDTH);
        labelLeaderboard.setPosition(SCREEN_WIDTH / 2f - labelLeaderboard.getWidth() / 2f, 150);
        labelLeaderboard.setAlignment(Align.center);
        labelLeaderboard.setWrap(true);

        addPressEffect(labelLeaderboard);

        buttonMusic = new Button(Assets.buttonStyleMusic);
        buttonMusic.setPosition(5, 5);
        buttonMusic.setChecked(!Settings.isMusicOn);
        buttonMusic.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                Settings.isMusicOn = !Settings.isMusicOn;
                buttonMusic.setChecked(!Settings.isMusicOn);
                if (Settings.isMusicOn)
                    Assets.playMusic();
                else
                    Assets.pauseMusic();
            }
        });

        buttonSound = new Button(Assets.buttonStyleSound);
        buttonSound.setPosition(75, 5);
        buttonSound.setChecked(!Settings.isSoundOn);
        buttonSound.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                Settings.isSoundOn = !Settings.isSoundOn;
                buttonSound.setChecked(!Settings.isSoundOn);
            }
        });

        buttonFacebook = new Button(Assets.buttonFacebook);
        buttonFacebook.setSize(50, 50);
        buttonFacebook.setPosition(SCREEN_WIDTH - buttonFacebook.getWidth() - 5, 10);
        addPressEffect(buttonFacebook);

        stage.addActor(titleImage);
        stage.addActor(labelPlay);
        stage.addActor(labelHelp);
        stage.addActor(labelLeaderboard);
        stage.addActor(labelRate);
        stage.addActor(buttonMusic);
        stage.addActor(buttonSound);
        stage.addActor(buttonFacebook);
    }

    @Override
    public void update(float delta) {

    }

    @Override
    public void draw(float delta) {
        batch.begin();
        batch.draw(Assets.backgroundAtlasRegion, 0, 0, SCREEN_WIDTH, SCREEN_HEIGHT);
        batch.end();
    }

    @Override
    public boolean keyDown(int keycode) {
        if (keycode == Keys.ESCAPE || keycode == Keys.BACK)
            Gdx.app.exit();
        return true;
    }
}

File: Two_Thousand_Fourty_Eight/core/src/com/nopalsoft/dosmil/screens/MainMenuScreen.kt
package com.nopalsoft.dosmil.screens

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Input
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Button
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.badlogic.gdx.utils.Align
import com.nopalsoft.dosmil.Assets
import com.nopalsoft.dosmil.Assets.pauseMusic
import com.nopalsoft.dosmil.Assets.playMusic
import com.nopalsoft.dosmil.MainGame
import com.nopalsoft.dosmil.Settings
import com.nopalsoft.dosmil.game.GameScreen

class MainMenuScreen(game: MainGame) : Screens(game) {
    private var titleImage: Image = Image(Assets.titleAtlasRegion)

    private var labelPlay: Label
    private var labelHelp: Label
    private var labelLeaderboard: Label
    private var labelRate: Label

    var buttonMusic: Button
    var buttonSound: Button
    private var buttonFacebook: Button

    init {
        titleImage.setPosition(SCREEN_WIDTH / 2f - titleImage.width / 2f, 580f)

        labelPlay = Label(Assets.languagesBundle!!["play"], Assets.labelStyleLarge)
        labelPlay.setPosition(SCREEN_WIDTH / 2f - labelPlay.width / 2f, 450f)
        addPressEffect(labelPlay)
        labelPlay.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                changeScreenWithFadeOut(GameScreen::class.java, game)
            }
        })

        // Help
        labelHelp = Label(Assets.languagesBundle!!["help"], Assets.labelStyleLarge)
        labelHelp.setPosition(SCREEN_WIDTH / 2f - labelHelp.width / 2f, 350f)
        addPressEffect(labelHelp)
        labelHelp.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                changeScreenWithFadeOut(HelpScreen::class.java, game)
            }
        })

        // Rate
        labelRate = Label(Assets.languagesBundle!!["rate"], Assets.labelStyleLarge)
        labelRate.setPosition(SCREEN_WIDTH / 2f - labelRate.width / 2f, 250f)
        addPressEffect(labelRate)

        // Leaderboard
        labelLeaderboard = Label(Assets.languagesBundle!!["leaderboard"], Assets.labelStyleLarge)
        labelLeaderboard.setFontScale(.85f)
        labelLeaderboard.width = SCREEN_WIDTH.toFloat()
        labelLeaderboard.setPosition(SCREEN_WIDTH / 2f - labelLeaderboard.width / 2f, 150f)
        labelLeaderboard.setAlignment(Align.center)
        labelLeaderboard.wrap = true

        addPressEffect(labelLeaderboard)

        buttonMusic = Button(Assets.buttonStyleMusic)
        buttonMusic.setPosition(5f, 5f)
        buttonMusic.isChecked = !Settings.isMusicOn
        buttonMusic.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                Settings.isMusicOn = !Settings.isMusicOn
                buttonMusic.isChecked = !Settings.isMusicOn
                if (Settings.isMusicOn) playMusic()
                else pauseMusic()
            }
        })

        buttonSound = Button(Assets.buttonStyleSound)
        buttonSound.setPosition(75f, 5f)
        buttonSound.isChecked = !Settings.isSoundOn
        buttonSound.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                Settings.isSoundOn = !Settings.isSoundOn
                buttonSound.isChecked = !Settings.isSoundOn
            }
        })

        buttonFacebook = Button(Assets.buttonFacebook)
        buttonFacebook.setSize(50f, 50f)
        buttonFacebook.setPosition(SCREEN_WIDTH - buttonFacebook.width - 5, 10f)
        addPressEffect(buttonFacebook)

        stage!!.addActor(titleImage)
        stage!!.addActor(labelPlay)
        stage!!.addActor(labelHelp)
        stage!!.addActor(labelLeaderboard)
        stage!!.addActor(labelRate)
        stage!!.addActor(buttonMusic)
        stage!!.addActor(buttonSound)
        stage!!.addActor(buttonFacebook)
    }

    override fun update(delta: Float) {
    }

    override fun draw(delta: Float) {
        batch!!.begin()
        batch!!.draw(Assets.backgroundAtlasRegion!!, 0f, 0f, SCREEN_WIDTH.toFloat(), SCREEN_HEIGHT.toFloat())
        batch!!.end()
    }

    override fun keyDown(keycode: Int): Boolean {
        if (keycode == Input.Keys.ESCAPE || keycode == Input.Keys.BACK) Gdx.app.exit()
        return true
    }
}
